pkgs/development/compilers/graalvm/default.nix

```diff
@@ -0,0 +1,203 @@
+{ stdenv, lib, fetchFromGitHub, fetchhg, fetchurl, mercurial, python27, zlib, makeWrapper, oraclejdk8 }:
+
+let
+  # pre-download some cache entries ('mx' will not be able to download under nixbld1)
+  makeMxCache = list:
+    stdenv.mkDerivation {
+      name = "mx-cache";
+      buildCommand = ''
+        mkdir $out
+        ${lib.concatMapStrings ({url, name, sha1}: ''
+                                  ln -s ${fetchurl { inherit url sha1; }} $out/${name}
+                                  echo -n ${sha1} > $out/${name}.sha1
+                                '') list}
+      '';
+    };
+
+  jvmci8-mxcache = [
+    rec { sha1 = "66215826a684eb6866d4c14a5a4f9c344f1d1eef"; name = "JACOCOCORE_${sha1}.jar";                            url = https://repo1.maven.org/maven2/org/jacoco/org.jacoco.core/0.7.9/org.jacoco.core-0.7.9.jar; }
```

Comment:
Have you generated those by hand?

Reply:
No, copied from the console on each download failure.

Reply:
Was there a particular reason to use `sha1` instead of `sha256`?

Reply:
I see, the project wants sha1.

Reply:
yes, it is nice that ```sha1``` is not yet deprecated in ```fetchurl```, otherwise in the table there would be both ```sha256``` and ```sha1```.